tedana/utils.py

```diff
@@ -108,6 +129,15 @@ def make_adaptive_mask(data, mask=None, getsum=False, threshold=1):
     # and count # of echos that pass criterion
     masksum = (np.abs(echo_means) > lthrs).sum(axis=-1)
 
+    # Determine where voxels stop decreasing in signal from echo to echo
+    echo_diffs = np.hstack((np.full((n_samples, 1), -1), np.diff(echo_means, axis=1)))
+    diff_mask = echo_diffs >= 0  # flag where signal is not decreasing
+    last_decreasing_echo = diff_mask.argmax(axis=1)
+    last_decreasing_echo[last_decreasing_echo == 0] = n_echos  # if no increase, set to n_echos
+
+    # Retain the more conservative of the two adaptive mask estimates
+    masksum = np.minimum(masksum, last_decreasing_echo)
```

Comment:
Might want to make a second warning message if `last_decreasing_echo` removes nontrivially more voxels than the other threshold. I'm viewing that as a sign that something is wrong with the data.

Reply:
I think we'll want to log the voxel counts with each different method. I wrote some code for that, but it was a bit ugly and I think it might be easier to do as a separate PR once all current adaptive mask PRs are merged.